backend/fms_core/utils.py

```diff
@@ -79,6 +79,14 @@ def str_cast_and_normalize(s) -> str:
     return str_normalize(str(s) if s is not None else s)
 
 
+def str_cast_and_normalize_lower(s) -> Union[str, None]:
+    """
+    Casts a value to a string, normalizes it and then converts to lower case.
+    """
+    result = str_cast_and_normalize(s)
+    return result.lower() if result is not None else result
```

Comment:
would be clearer with "else None" as it clearly state the return value

Reply:
I hesitated between returning None or result but I agree - returning None would be more readable.